File: samples/java17-sample-contract/src/main/java/my/sample/java17/contract/rockps/api/RockPaperScissorsContract.java
package my.sample.java17.contract.rockps.api;

import com.wavesenterprise.sdk.contract.api.annotation.ContractAction;
import com.wavesenterprise.sdk.contract.api.annotation.ContractInit;
import my.sample.java17.contract.rockps.game.request.CreateGameRequest;
import my.sample.java17.contract.rockps.game.request.PlayRequest;
import my.sample.java17.contract.rockps.game.request.RevealRequest;

public interface RockPaperScissorsContract {

    @ContractInit
    void createGame(CreateGameRequest createGameRequest);

    @ContractAction
    void play(PlayRequest registerPlayerRequest);

    @ContractAction
    void reveal(RevealRequest revealRequest);

    class Keys {
        public static final String PLAYER_ADDRESSES_KEY = "PLAYER_ADDRESSES";
        public static final String PLAYERS_MAPPING_PREFIX = "PLAYERS";

        public final static String GAME_KEY = "GAME";
        public static final String GAME_STATUS_KEY = "GAME_STATUS";
        public static final String GAME_WINNER_ADDR_KEY = "GAME_WINNER_ADDRESS";

        public static final String CREATED_DATE_KEY = "CREATED_DATE";
    }
}

File: samples/kotlin-sample-contract/src/main/kotlin/my/sample/kotlin/contract/rockps/api/RockPaperScissorsContract.kt
package my.sample.kotlin.contract.rockps.api

import com.wavesenterprise.sdk.contract.api.annotation.ContractAction
import com.wavesenterprise.sdk.contract.api.annotation.ContractInit
import my.sample.kotlin.contract.rockps.game.request.CreateGameRequest
import my.sample.kotlin.contract.rockps.game.request.PlayRequest
import my.sample.kotlin.contract.rockps.game.request.RevealRequest

interface RockPaperScissorsContract {

    @ContractInit
    fun createGame(createGameRequest: CreateGameRequest)

    @ContractAction
    fun play(registerPlayerRequest: PlayRequest)

    @ContractAction
    fun reveal(revealRequest: RevealRequest)

    object Keys {
        const val PLAYER_ADDRESSES_KEY = "PLAYER_ADDRESSES"
        const val PLAYERS_MAPPING_PREFIX = "PLAYERS"
        const val GAME_KEY = "GAME"
        const val GAME_STATUS_KEY = "GAME_STATUS"
        const val GAME_WINNER_ADDR_KEY = "GAME_WINNER_ADDRESS"
        const val CREATED_DATE_KEY = "CREATED_DATE"
    }
}
